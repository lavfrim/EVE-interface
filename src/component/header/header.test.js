import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';

import Header from './index';

const mockStore = configureMockStore();
const storeWithMessage = mockStore({
    errorMessage: 'error message',
});
const storeWithoutMessage = mockStore({
    errorMessage: '',
});

describe('header should render correctly', () => {
    it('with error mesage', () => {
        const wrapper = mount(
            <Header store={storeWithMessage} />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('without error mesage', () => {
        const wrapper = mount(
            <Header store={storeWithoutMessage} />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});


// it('CHANGE_TEST', () => {
//     expect(1).toEqual(1);
// })
