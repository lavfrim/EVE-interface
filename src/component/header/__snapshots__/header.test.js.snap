// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should render correctly with error mesage 1`] = `
<Connect(Header)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Header
    dispatch={[Function]}
    errorMessage="error message"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <header
      className="header"
    >
      <p>
        EVE interface / interview task
      </p>
      <p>
        error message
      </p>
    </header>
  </Header>
</Connect(Header)>
`;

exports[`header should render correctly without error mesage 1`] = `
<Connect(Header)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Header
    dispatch={[Function]}
    errorMessage=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <header
      className="header"
    >
      <p>
        EVE interface / interview task
      </p>
    </header>
  </Header>
</Connect(Header)>
`;
